### PR TITLE
Add extension type for profile report range JSON

### PR DESCRIPTION
**Pull request by srawlins**
An alternative to some of the code in https://github.com/flutter/devtools/pull/6948

Work towards https://github.com/flutter/devtools/issues/6929

This one is, I think, a fairly straightforwards extension type. It contains the safe access of JSON fields, and the const field names. We wrap access to the JSON as early as possible, at the very end of the diff, line ~1069, before we enter `ProfileReportRange._fromJson`. Other notes:

* _ProfileReportRangeJson is private. I think this makes sense; it is only accessed from this library.
* The representation field in _ProfileReportRangeJson is public (`json`). Shrug. It could be private. Maybe it should be.
* The `_profile` field, the outermost access on `json` is private, because it is only ever accessed _from within the extension type_. But if this clashes with local style, we can change.
* The other extension type fields are public.
* No verification is done. It is possible, and maybe desirable, to have a constructor that verifies the structure of the JSON. But I don't know what that structure should be (aside from the fields used), so I didn't add any verification.

Other cleanups here:

* ProfileReportMetaData had one field, was only accessed within the same library, and was only accessed to access that field; removed.
* None of the `metadata`, `inclusiveTicks`, `exclusiveTicks`, or `list` fields on ProfileReportRange were accessed outside the constructor. So, removed.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or there is a reason for not adding tests.

![build.yaml badge]

If you need help, consider asking for help on [Discord].

[Contributor Guide]: https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[build.yaml badge]: https://github.com/flutter/devtools/actions/workflows/build.yaml/badge.svg

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart [hunk 1]
```diff
@@ -1010,47 +1005,58 @@ class ProfileReportEntry {
...
-              // either be a negative value or a String (e.g., 'ParallelMove'
-              // or 'NoSource'). We'll just use -1 as a placeholder since we
-              // won't display anything for these tokens anyway.
-              (e) => e is int
-                  ? script.getLineNumberFromTokenPos(e) ?? _kNoSourcePosition
-                  : _kNoSourcePosition,
-            )
-            .toList() {
+  ProfileReportRange._fromJson(Script script, _ProfileReportRangeJson json) {
+    final sampleCount = json.metadata[_kSampleCountKey] as int;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart:1009]: should sampleCount be a getter in the extension type?

File: packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart [hunk 2]
```diff
@@ -1010,47 +1005,58 @@ class ProfileReportEntry {
...
-        sampleCount: metadata.sampleCount,
+        sampleCount: sampleCount,
         line: line,
         inclusive: inclusiveTicks[i],
         exclusive: exclusiveTicks[i],
       );
     }
   }
 
+  static const _kSampleCountKey = 'sampleCount';
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart:1037]: if we add a sampleCount getter below, this const could move into the extension type too

**Review by kenzieschmoll — APPROVED** [review]

File: packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart (hunk 2 above)
Comment (srawlins) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart]: Done

File: packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart (hunk 1 above)
Comment (srawlins) [reply packages/devtools_app/lib/src/screens/vm_developer/vm_service_private_extensions.dart]: done

**Pull request merged**